xilem/examples/elm.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use masonry::widget::{CrossAxisAlignment, MainAxisAlignment};
+use winit::error::EventLoopError;
+use xilem::{
+    view::{button, flex, label},
+    EventLoop, WidgetView, Xilem,
+};
+use xilem_core::{adapt, map_action, map_state, MessageResult};
+
+#[derive(Default)]
+struct AppState {
+    adapt_count: i32,
+    map_state_count: i32,
+    map_action_count: i32,
+}
+
+// `map_state()` maps a subset of the state from the parent, such that views can be modularized by state
+fn map_state_view(count: &mut i32) -> impl WidgetView<i32> {
```

Comment:
This would not be the idiomatic name for this view; maybe something like `direct_counter`?